src/mkdir.js

```diff
@@ -19,17 +19,19 @@ function mkdirSyncRecursive(dir) {
     common.error('dirname() failed: [' + dir + ']');
   }
 
-  // Base dir exists, no recursion necessary
-  if (fs.existsSync(baseDir)) {
-    fs.mkdirSync(dir, parseInt('0777', 8));
-    return;
-  }
-
   // Base dir does not exist, go recursive
-  mkdirSyncRecursive(baseDir);
+  if (!fs.existsSync(baseDir)) {
+    mkdirSyncRecursive(baseDir);
+  }
 
-  // Base dir created, can create dir
-  fs.mkdirSync(dir, parseInt('0777', 8));
+  try {
+    // Base dir created, can create dir
+    fs.mkdirSync(dir, parseInt('0777', 8));
+  } catch (e) {
+    // swallow directory exists errors ('EEXIST'); mitigates race conditions
+    /* istanbul ignore next */
```

Comment:
I think we can remove `istanbul ignore next` because this now handles the non-racy condition where base dir exists.

You could also update the comment since this isn't only for race conditions (it's just the non-racy way to create a directory if it doesn't already exist).

Reply:
Done.

Reply:
It looks like this caught some missing test coverage 😄  I think we weren't testing the case where `mkdir -p` is called on a directory which already exists. I filed issue #1022 to track this.